packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -1125,32 +1104,30 @@ export class PipelineEditor extends React.Component<
    * @returns null if pipeline is valid, error message if not.
    */
   validateAllLinks(): string {
```

Comment:
Thank you for starting this cleanup, at some point I really would like to see one call in the widget, which is PipelineService.validatePipeline(....) and everything else is driven from there (with calling other utility classes).

Reply:
👍 yea that's how it is in the large PR